File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        

        dic = defaultdict(int)
        nums = set() 
        for a, b in matches: 
            dic[b] += 1 
            nums.add(a) 
            nums.add(b)
        print(dic) 

        left = [] 
        right = [] 
        for i in nums: 

            if dic[i] ==1: 
                right.append( i)
            elif dic[i] ==0: 
                left.append(i)
        
        left.sort() 
        right.sort() 
        ans = [] 
        ans.append(left)
        ans.append(right) 

        return ans 

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        

        int[] win = new int[100001];
        int[] loss = new int[100001]; 

        for(int[] mc: matches){
            win[mc[0]]++; 
            loss[mc[1]]++;
        }

        List<List<Integer>> ans = new ArrayList<>(); 
        List<Integer> left = new ArrayList<>(); 
        List<Integer> right = new ArrayList<>();

        for(int i =0; i < win.length ; i++){
            if( win[i] > 0 && loss[i] == 0 ){
                left.add(i); 
            }
            if(loss[i] ==1){
                right.add(i);
            }
        }
        ans.add(left); 
        ans.add(right); 

        return ans; 


    }
}